### PR TITLE
Order endpoints

### PR DESCRIPTION
**Pull request by akjanik**
I want to merge this change because it adds ability to query orders with API.
Note: it require changes from
https://github.com/mirumee/django-prices/issues/61
so build will fail until above PR is merged and released.

### Pull Request Checklist

(Please keep this section. It will make maintainer's life easier.)

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

File: saleor/graphql/order/filters.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from django_filters import (
+    CharFilter, DateFromToRangeFilter, NumberFilter)
+from graphene_django.filter.filterset import GlobalIDMultipleChoiceFilter
+
+from ...order import models
+from ..core.filters import DistinctFilterSet
+
+
+class OrderFilter(DistinctFilterSet):
+    """
```
Comment (patrys) [inline saleor/graphql/order/filters.py]: This is not a valid docstring, please test with `pydocstyle`

File: saleor/graphql/order/resolvers.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from .types import Order
+
+
+def resolve_orders(info):
+    user = info.context.user
+    if user.is_anonymous:
+        raise PermissionDenied('You have no permission to see this')
+    if user.get_all_permissions() & {'order.view_order', 'order.edit_order'}:
+        return models.Order.objects.all().distinct()
+    return models.Order.objects.filter(user=user).distinct()
```
Comment (elwoodxblues) [inline saleor/graphql/order/resolvers.py]: If I'm correct this queryset would be used for example on customer's account page in the storefront to show his orders, right? In the current account view we use this queryset: `user.orders.confirmed()`. We could reuse it here to be consistent.

File: saleor/graphql/order/resolvers.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def resolve_order(info, id):
+    order = get_node(info, id, only_type=Order)
+    user = info.context.user
+    if (order.user == user
+        or user.is_superuser
```
Comment (elwoodxblues) [inline saleor/graphql/order/resolvers.py]: Do we have to explicitly check `is_superuser`? I think `get_all_permissions` would return all permissions anyway if the user is a superuser.

File: saleor/graphql/order/resolvers.py (hunk 2 above)
Comment (akjanik) [reply saleor/graphql/order/resolvers.py]: Correct. I will add some docstrings explaining it and take a look at this queryset.

File: saleor/graphql/order/types.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+class Order(CountableDjangoObjectType):
+    order_id = graphene.Int(
+        description='Order number.')
+
+    class Meta:
+        description = 'Represents an order in the shop.'
+        interfaces = [relay.Node]
+        model = models.Order
+        exclude_fields = []
```
Comment (elwoodxblues) [inline saleor/graphql/order/types.py]: We should exclude `total_net` and `total_gross` in favor of having only `total` field to access both values. The same applies to the `shipping_price` fied. `TaxedMoney` type should also give access to `tax` property.

**Pull request merged**